### PR TITLE
Increase move verifier timeout & add metrics

### PR DESCRIPTION
**Pull request by oxade**
## Description 

Increases the verifier timeout limits, adds test
Adds metrics and metrics tests

## Test Plan 

Unit tests

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/metrics.rs [hunk 1]
```diff
@@ -69,3 +69,27 @@ impl LimitsMetrics {
...
+    pub verifier_timeout_metrics: IntCounterVec,
+}
+
+impl BytecodeVerifierMetrics {
+    pub const MOVE_VERIFIER_TAG: &'static str = "move_verifier";
+    pub const SUI_VERIFIER_TAG: &'static str = "sui_verifier";
+    pub const OVERALL_TAG: &'static str = "overall";
+    pub const SUCCESS_TAG: &'static str = "success";
+    pub const TIMEOUT_TAG: &'static str = "failed";
+    pub fn new(registry: &prometheus::Registry) -> Self {
```
Comment (amnn) [inline crates/sui-types/src/metrics.rs:84]: What's the pro's/con's on this approach vs having separate counters for each of these states?

File: crates/sui-adapter/src/adapter.rs [hunk 2]
```diff
@@ -214,13 +218,18 @@ pub fn run_metered_move_bytecode_verifier_impl(
...
             {
+                metrics.verifier_timeout_metrics.with_label_values(&[BytecodeVerifierMetrics::MOVE_VERIFIER_TAG, BytecodeVerifierMetrics::TIMEOUT_TAG]).inc();
                 return Err(SuiError::ModuleVerificationFailure {
                     error: "Verification timedout".to_string(),
                 });
             };
         } else {
-            sui_verify_module_metered(protocol_config, module, &BTreeMap::new(), meter)?
+            sui_verify_module_metered(protocol_config, module, &BTreeMap::new(), meter).map_err(|err|{
+                metrics.verifier_timeout_metrics.with_label_values(&[ BytecodeVerifierMetrics::SUI_VERIFIER_TAG, BytecodeVerifierMetrics::TIMEOUT_TAG]).inc();
```
Comment (amnn) [inline crates/sui-adapter/src/adapter.rs:228]: Note that these are not all timeouts, currently.

**Review by amnn — APPROVED** [review]
I think it might be slightly neater to implement the metrics as individual counters:

- `move_verifier_timeout`
- `sui_verifier_failed`
- `verifier_succeeded`

But otherwise, LGTM, thanks @oxade !

File: crates/sui-adapter/src/adapter.rs [hunk 3]
```diff
@@ -214,13 +218,18 @@ pub fn run_metered_move_bytecode_verifier_impl(
...
+                err
+            })?
         }
+        metrics.verifier_timeout_metrics.with_label_values(&[BytecodeVerifierMetrics::OVERALL_TAG, BytecodeVerifierMetrics::SUCCESS_TAG]).inc();
```
Comment (amnn) [inline crates/sui-adapter/src/adapter.rs:232]: Do we also want to add the timing metric here?

File: crates/sui-types/src/metrics.rs (hunk 1 above)
Comment (oxade) [reply crates/sui-types/src/metrics.rs:84]: This was just the personal preference of folks I consulted on this.

File: crates/sui-adapter/src/adapter.rs (hunk 2 above)
Comment (oxade) [reply crates/sui-adapter/src/adapter.rs:228]: Yes. But for this purpose we treat the failure as a timeout pending when we disambiguate.

File: crates/sui-adapter/src/adapter.rs (hunk 3 above)
Comment (oxade) [reply crates/sui-adapter/src/adapter.rs:232]: Yes that will come in next PR.

File: crates/sui-adapter/src/adapter.rs (hunk 2 above)
Comment (amnn) [reply crates/sui-adapter/src/adapter.rs:228]: Got it, is that also a "soon" follow-up? (like the timing PR?)

**Pull request merged**